src/test/java/qna/domain/QuestionRepositoryTest.java

```diff
@@ -18,17 +16,35 @@ class QuestionRepositoryTest {
 
     @Autowired
     QuestionRepository questionRepository;
+    @Autowired
+    UserRepository userRepository;
+
 
     @Test
     void save() {
-        final Question actual = questionRepository.save(QuestionTest.Q3);
+        final User writer = userRepository.save(UserTest.TESTUSER);
+        final Question actual = questionRepository.save(QuestionTest.Q3.writeBy(writer));
+        //final Question actual = questionRepository.save(QuestionTest.Q1.writeBy(writer));
         assertThat(actual.getId()).isEqualTo(QuestionTest.Q3.getId());
-    }
+    }// db에 이제 Q1이 지닌 상태
+
+    @Test
+    @DisplayName("save 학습테스트. 같은 객체를 중복으로 저장하였을 때 결과 확인")
+    void duplicateSave() {
+        final User writer = userRepository.save(UserTest.TESTUSER);
+        final Question question1 = questionRepository.save(QuestionTest.Q1.writeBy(writer)); //db 에는 없고 persist 로 영속성에만 저장
+        final Question question2 = questionRepository.save(QuestionTest.Q1.writeBy(writer)); //db 에는 없고 persist 로 영속성에만 저장
+        assertThat(question1).isEqualTo(question2); //같은 객체인지 확인
+        assertThat(question1.getId()).isEqualTo(question2.getId());
+        assertThat(question1.getId()).isEqualTo(QuestionTest.Q1.getId());
+    }// test 하나 끝나면 commit 되어 db에 반영
```

Comment:
```suggestion
    @Test
    @DisplayName("save 학습테스트. 같은 객체를 중복으로 저장하였을 때 결과 확인")
    void duplicateSave() {
        final User writer = userRepository.save(UserTest.TESTUSER);
        final Question question1 = questionRepository.save(QuestionTest.Q3.writeBy(writer)); //db 에는 없고 persist 로 영속성에만 저장
        final Question question2 = questionRepository.save(QuestionTest.Q3.writeBy(writer)); //db 에는 없고 persist 로 영속성에만 저장
        assertThat(question1).isEqualTo(question2); //같은 객체인지 확인
        assertThat(question1.getId()).isEqualTo(question2.getId());
        assertThat(question1.getId()).isEqualTo(QuestionTest.Q3.getId());
    }// test 하나 끝나면 commit 되어 db에 반영
```

이 테스트는 테스트 순서에 따라 실패하는 경우가 발생하곘네요.
위와 같이 변경했을 때 왜 성공할까요?

Reply:
데일리미팅을 통해서 다시한번 숙지하였습니다! 감사합니다! 😄